Vagrantfile

```diff
@@ -58,20 +58,27 @@ Vagrant.configure("2") do |config|
         config.vagrant.plugins = required_plugins
     end
 
-	config.vm.synced_folder ".", "/vagrant", disabled: false, type: 'rsync'
+    config.vm.synced_folder '.', configuration['base_folder'], disabled: false, type: 'rsync'
+    config.vm.provision 'file', source: 'configuration.yml', destination: configuration['base_folder'] + '/web_s2i/configuration.yml'
 
   config.vm.provision "docker" do |d|
-    d.build_image "/vagrant/docker",
-      args: "-t devopsloft/devopsloft"
+    d.post_install_provision "shell", inline:"docker network create devopsloft_network"
+    d.build_image configuration['base_folder'] + '/db_s2i',
+      args: '-t ' + configuration['namespace'] + '/' + configuration['dockerhub_db'] + ' --build-arg MYSQL_ROOT_PASSWORD=' + configuration['mysql_root_password'] + ' --build-arg MYSQL_DATABASE=' + configuration['mysql_db']
+    d.run "db",
```

Comment:
This whole block seems very complicated to me, too many references.
A. If the image is built locally, the dockerhub has no meaning, the image name just needs to be consistent, In addition, docker supports `--env_file`, it's not yaml, but, it's cleaner and more DRY.

Reply:
I agree, 
Is there a way of using `--env_file`,  with variables inside ?
I am aiming to have only one place of environment parameters.

Reply:
`--env_file` like so https://docs.docker.com/compose/compose-file/#env_file
Note that it is **not** a shell script, and variable substitution there doesn't work.
you can also use it in vagrant, see https://www.nickhammond.com/configuring-vagrant-virtual-machines-with-env/, and it's also supported later with docker-compose

Reply:
Still, it does not answer my wish:
To use one file for defining variables for vagrant, docker, application.py and .travis.yml

Reply:
ok, done with .env